File: python/anagram_checker.py
print("Enter two strings and I'll tell you if they are anagrams:")

first_string = input("Enter the first string: ")
second_string = input("Enter the second string: ")


def is_anagram(string1: str, string2: str) -> bool:
    # check if the strings have same length
    if len(string1) != len(string2):
        return False

    for i in range(len(string1)):
        if string1[i] not in string2:
            return False

    return True


if is_anagram(first_string, second_string):
    print(f"{first_string} and {second_string} are anagrams.")
else:
    print(f"{first_string} and {second_string} are not anagrams.")

File: go/anagram_checker.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)

	fmt.Println("Enter two strings and I'll tell you if they are anagrams:")

	fmt.Print("Enter the first string: ")
	scanner.Scan()
	firstString := scanner.Text()

	fmt.Print("Enter the second string: ")
	scanner.Scan()
	secondString := scanner.Text()

	if isAnagram(firstString, secondString) {
		fmt.Printf("%s and %s are anagrams.\n", firstString, secondString)
	} else {
		fmt.Printf("%s and %s are not anagrams.\n", firstString, secondString)
	}
}

func isAnagram(string1 string, string2 string) bool {
	// check if the strings have same length
	if len(string1) != len(string2) {
		return false
	}

	for _, val := range string1 {
		if !strings.Contains(string2, string(val)) {
			return false
		}
	}

	return true
}
